### ISSUE TITLE
Incorrect error "E2523 Only one of [VirtualName, Ebs] should be specified for ..."

### ISSUE THREAD

**Issue by lorengordon** [awslabs/cfn-python-lint#255]
*cfn-lint version: (`cfn-lint --version`)*

```
$ cfn-lint --version
cfn-lint 0.4.2
```

*Description of issue.*

cfn-lint is outputting an error about a property that we aren't using, `VirtualName`. Not really sure what's up. `VirtualName` is not present _anywhere_ in the template... Here's the error:

```
E2523 Only one of [VirtualName, Ebs] should be specified for ...
```

Here is the relevant part of the template...

```
                "BlockDeviceMappings": [
                    {
                        "DeviceName": {
                            "Fn::Join": [
                                "",
                                [
                                    "/dev/",
                                    {
                                        "Fn::FindInMap": [
                                            "Distro2RootDevice",
                                            {
                                                "Ref": "AmiDistro"
                                            },
                                            "DeviceName"
                                        ]
                                    }
                                ]
                            ]
                        },
                        "Ebs": {
                            "DeleteOnTermination": "true",
                            "VolumeType": "gp2"
                        }
                    }
                ]
```

**Comment by kddejong** [comment]
I'm having troubles repeating this.  I may need some more information for you but I will do some more digging.   Error could use a little bit better wording but you will see this error under two different scenarios:
- Neither Ebs or VirtualName are specified
- Both Ebs and VirtualName are specified

When I put your template snippets into my files I don't get the error.  So trying to play out a few more scenarios that could give you this error.

**Comment by lorengordon** [comment]
The full templates are a bit complex. Check it out here:

* https://github.com/plus3it/terraform-aws-watchmaker/blob/master/modules/lx-instance/watchmaker-lx-instance.template.cfn.json#L1343

You can see the failure in the travis-ci build where I'm trying to integrate cfn-lint...

* https://travis-ci.org/plus3it/terraform-aws-watchmaker/jobs/411697708

**Comment by lorengordon** [comment]
I'm guessing it's the former scenario that's triggering, with the conditional second volume... sorry about that, I was trying to simplify the issue report and didn't think about the conditional second volume perhaps triggering the problem...

* Neither Ebs or VirtualName are specified

```
                "BlockDeviceMappings": [
                    {
                        "DeviceName": {
                            "Fn::Join": [
                                "",
                                [
                                    "/dev/",
                                    {
                                        "Fn::FindInMap": [
                                            "Distro2RootDevice",
                                            {
                                                "Ref": "AmiDistro"
                                            },
                                            "DeviceName"
                                        ]
                                    }
                                ]
                            ]
                        },
                        "Ebs": {
                            "DeleteOnTermination": "true",
                            "VolumeType": "gp2"
                        }
                    },
                    {
                        "Fn::If": [
                            "CreateAppVolume",
                            {
                                "DeviceName": "/dev/xvdf",
                                "Ebs": {
                                    "DeleteOnTermination": "true",
                                    "VolumeSize": {
                                        "Ref": "AppVolumeSize"
                                    },
                                    "VolumeType": {
                                        "Ref": "AppVolumeType"
                                    }
                                }
                            },
                            {
                                "Ref": "AWS::NoValue"
                            }
                        ]
                    }
],
```

**Comment by kddejong** [comment]
Got it.  I can fix that one :) Thanks
